applications/PfemFluidDynamicsApplication/python_scripts/remesh_fluid_domains_process.py

```diff
@@ -165,18 +166,19 @@ def ExecuteInitializeSolutionStep(self):
 
         if(currentStep > 1):
             for domain in self.meshing_domains:
-                domain.ComputeAverageMeshParameters()  
-                meanVolumeBeforeMeshing=domain.GetMeanVolume()
-                totalVolumeBeforeMeshing=domain.GetTotalVolume()
+                if(domain.Active()):
+                    domain.ComputeAverageMeshParameters()  
+                    meanVolumeBeforeMeshing=domain.GetMeanVolume()
+                    totalVolumeBeforeMeshing=domain.GetTotalVolume()
 
-                fileTotalVolume = open("totalVolumeBeforeMeshing.ods", 'a')
-                if(currentStep==2):
-                    fileTotalVolume.seek(0)
-                    fileTotalVolume.truncate()
+                    fileTotalVolume = open("totalVolumeBeforeMeshing.ods", 'a')
```

Comment:
+1 for the comment by @jcotela. However, this branch was opened to fix an error. I just checked if the error is fixed and it is. So I will approve the merge and I would suggest to open a new branch to optimize this piece of code.